### PR TITLE
update models and migrations for Django 2.0 compatibility

### PR DESCRIPTION
**Pull request by safonin**
update models and migrations for Django 2.0 compatibility

### CODE REVIEW

File: knox/models.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class AuthToken(models.Model):
     salt = models.CharField(
         max_length=CONSTANTS.SALT_LENGTH, unique=True)
     user = models.ForeignKey(
-        User, null=False, blank=False, related_name='auth_token_set')
+        User, null=False, blank=False, related_name='auth_token_set', on_delete=models.CASCADE))
```
Comment (bufke) [inline knox/models.py]: There's an added `)` here that causes Travis CI to fail.

**Comment by safonin** [comment]
sorry, fixed

**Comment by belugame** [comment]
Thank you for your contribution @safonin @bufke 
I will prepare the travis metadata to drop django 1.8 and build 1.11 & 2.0 instead.

**Comment by belugame** [comment]
develop branch now builds django 1.11 and 2.0 via tox/travis. For 2.0 there are test failures. Might be an easy fix, happy for someone helping finding the issue so we can release soon!

https://travis-ci.org/James1345/django-rest-knox/builds/316640570?utm_source=github_status&utm_medium=notification

**Comment by belugame** [comment]
@safonin @bufke @philngo
I just released version 3.1.0 to pypi with Django 2.0 support thanks to @vkylamba!

**Pull request merged**